### PR TITLE
CSPL-1682: Splunk pods get recycled after upgrading operator from 1.0…

### PR DESCRIPTION
**Pull request by vivekr-splunk**
- changes done secret comparison before updating ownership reference. 
- set requeueafter to 0 when requeue is false.

### CODE REVIEW

**Review by smohan-splunk — APPROVED** [review]

File: pkg/splunk/controller/configmap.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func SetConfigMapOwnerRef(ctx context.Context, client splcommon.ControllerClient
 	currentOwnerRef := configMap.GetOwnerReferences()
 	// Check if owner ref exists
 	for i := 0; i < len(currentOwnerRef); i++ {
-		if reflect.DeepEqual(currentOwnerRef[i], splcommon.AsOwner(cr, false)) {
+		if reflect.DeepEqual(currentOwnerRef[i].UID, cr.GetUID()) {
```
Comment (sgontla) [inline pkg/splunk/controller/configmap.go:133]: You mentioned that we can include the Kind as well. In that line of thought, just wondering, is it a good idea, just to exclude the API version alone?  A new API that compares all the fields except API version.
```
`// AsOwner returns an object to use for Kubernetes resource ownership references.
func AsOwner(cr MetaObject, isController bool) metav1.OwnerReference {
	return metav1.OwnerReference{
		APIVersion: cr.GetObjectKind().GroupVersionKind().GroupVersion().String(),
		Kind:       cr.GetObjectKind().GroupVersionKind().Kind,
		Name:       cr.GetObjectMeta().GetName(),
		UID:        cr.GetObjectMeta().GetUID(),
		Controller: &isController,
	}
}`
```

File: pkg/splunk/controller/statefulset.go [hunk 2]
```diff
@@ -326,7 +326,7 @@ func DeleteReferencesToAutomatedMCIfExists(ctx context.Context, client splcommon
 
 //isCurrentCROwner returns true if current CR is the ONLY owner of the automated MC
 func isCurrentCROwner(cr splcommon.MetaObject, currentOwners []metav1.OwnerReference) bool {
-	return reflect.DeepEqual(currentOwners[0], splcommon.AsOwner(cr, false))
+	return reflect.DeepEqual(currentOwners[0].UID, cr.GetUID())
```
Comment (akondur) [inline pkg/splunk/controller/statefulset.go:329]: Same comment as above. Maybe we can create function to compare the four fields - kind, name, UID and controller.

File: pkg/splunk/util/secrets.go [hunk 3]
```diff
@@ -121,7 +121,7 @@ func SetSecretOwnerRef(ctx context.Context, client splcommon.ControllerClient, s
 	currentOwnerRef := secret.GetOwnerReferences()
 	// Check if owner ref exists
 	for i := 0; i < len(currentOwnerRef); i++ {
-		if reflect.DeepEqual(currentOwnerRef[i], splcommon.AsOwner(cr, false)) {
+		if reflect.DeepEqual(currentOwnerRef[i].UID, cr.GetUID()) {
```
Comment (akondur) [inline pkg/splunk/util/secrets.go:124]: Same comment as above.

File: pkg/splunk/controller/configmap.go (hunk 1 above)
Comment (vivekr-splunk) [reply pkg/splunk/controller/configmap.go:133]: we do not need controller, we just need  UID. UID will uniquely identify the resource. please [refer](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#:~:text=A%20Kubernetes%20systems%2Dgenerated%20string,(also%20known%20as%20UUIDs).).

File: pkg/splunk/controller/statefulset.go (hunk 2 above)
Comment (vivekr-splunk) [reply pkg/splunk/controller/statefulset.go:329]: we do not need controller, we just need  UID. UID will uniquely identify the resource. please [refer](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#:~:text=A%20Kubernetes%20systems%2Dgenerated%20string,(also%20known%20as%20UUIDs).).

File: pkg/splunk/controller/configmap.go (hunk 1 above)
Comment (akondur) [reply pkg/splunk/controller/configmap.go:133]: Does the UID remain the if we update the name, kind of the CR(I'm not sure if these are even possible)? If yes then comparing the UID's should be enough.

**Review by sgontla — APPROVED** [review]

**Review by akondur — APPROVED** [review]

**Pull request merged**